File: packages/typia/src/transformers/features/http/CreateHttpAssertQueryTransformer.ts
import { HttpAssertQueryProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace CreateHttpAssertQueryTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.factory({
      ...props,
      method: "http.createAssertQuery",
      write: HttpAssertQueryProgrammer.write,
    });
}

File: packages/transform/src/features/http/CreateHttpAssertQueryTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var CreateHttpAssertQueryTransformer = createHttpAssertQueryTransformerNamespace{}

type createHttpAssertQueryTransformerNamespace struct{}

func (createHttpAssertQueryTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpAssertQueryProgrammer.Write(call.TypeText)
}
